### PR TITLE
feat(request): add request provider for SSR without adding express as…

### PR DESCRIPTION
**Pull request by valorkin**
… a dep to a client lib

### CODE REVIEW

**Comment by pavankjadda** [comment]
@valorkin thanks for the PR. Would you mind explaining the process you followed i.e. what does new package `express-ssr` do and instructions to test this PR

File: projects/ngx-cookie-service/src/lib/cookie.service.ts [hunk 1]
```diff
@@ -14,7 +15,8 @@ export class CookieService {
   constructor(
     @Inject(DOCUMENT) private document: Document,
     // Get the `PLATFORM_ID` so we can check if we're in a browser.
-    @Inject(PLATFORM_ID) private platformId
+    @Inject(PLATFORM_ID) private platformId: any,
+    @Optional() @Inject(REQUEST_PROVIDER_TOKEN) private request: RequestProvider
```
Comment (pavankjadda) [inline projects/ngx-cookie-service/src/lib/cookie.service.ts:19]: @valorkin I see you created `Injection Token` and supplied it here. Does it work for CSR along with SSR application? Would mind creating a sample repo that we can use to test this. 

@stevermeister take a look at this

File: projects/ngx-cookie-service/express-ssr/src/express-request.provider.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import {StaticProvider} from '@angular/core';
+import {REQUEST} from '@nguniversal/express-engine/tokens';
+import {REQUEST_PROVIDER_TOKEN, RequestProvider} from 'ngx-cookie-service/request-provider';
+
+export const requestProviderFactory = (request: RequestProvider) => {
+  return request;
+};
+
+export const expressRequestProvider: StaticProvider = {
```
Comment (pavankjadda) [inline projects/ngx-cookie-service/express-ssr/src/express-request.provider.ts:9]: To be honest I have very little knowledge on this. I will read more about this

**Comment by valorkin** [comment]
@pavankjadda you can discuss it with @stevermeister

File: projects/ngx-cookie-service/src/lib/cookie.service.ts (hunk 1 above)
Comment (stevermeister) [reply projects/ngx-cookie-service/src/lib/cookie.service.ts:19]: yes, now playing with it here - https://github.com/stevermeister/ssr-coookie

**Comment by pavankjadda** [comment]
Closes in favor of #237

**Pull request closed without merging**